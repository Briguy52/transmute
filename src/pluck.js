import curry from "./curry";
import get from "./get";

/**
 * Select `key` from each item in `subject`.
 *
 * @example
 * const pluckName = pluck('name');
 * pluckName(userMap) === Map({123: 'Testing'});
 *
 * @param  {any} key
 * @param  {Iterable} subject
 * @return {Iterable}
 */
function pluck(key, subject) {
  return subject.map(get(key));
}

export default curry(pluck);
