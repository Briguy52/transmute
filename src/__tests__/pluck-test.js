import { List, Map } from "immutable";
import pluck from "../pluck";

describe("transmute/pluck", () => {
  it("extracts the value of each key in each item", () => {
    expect(
      pluck("n", List.of(Map({ n: 1 }), Map({ n: 2 }), Map({ n: 3 })))
    ).toMatchSnapshot();
  });
});
