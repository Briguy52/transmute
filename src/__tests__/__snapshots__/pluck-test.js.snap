// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/pluck extracts the value of each key in each item 1`] = `
Immutable.List [
  1,
  2,
  3,
]
`;
